flow/flow.py

```diff
@@ -1036,12 +1036,20 @@ def firebase_send_sensor_data(self, firebase, send_sensor_data):
         # Enable https connection reuse to Firebase
         session = requests.Session()
         session.mount('https://', requests.adapters.HTTPAdapter())
-
+						
         # Exchange the custom token for an id token
-        r = session.post(google_auth_url, headers={'Content-Type': 'application/json'}, data = json.dumps({'token': firebase["token"], 'returnSecureToken': True}))
-        if r.status_code != 200:
-            logging.error('ABORTING firebase_send_sensor_data thread!  POST to %s returned %s' % (google_auth_url, r.status_code))
-            return
+        post_succeeded = False;
+        while not post_succeeded:
+            try:
+                r = session.post(google_auth_url, headers={'Content-Type': 'application/json'}, data = json.dumps({'token': firebase["token"], 'returnSecureToken': True}))
+                if r.status_code != 200:
+                    logging.error('ABORTING firebase_send_sensor_data thread!  POST to %s returned %s' % (google_auth_url, r.status_code))
+                    return
+                post_succeeded = True
+            except Exception as err:
+                logging.error("Session post error: %s" % (err))
+                c.sleep(0.1)
```

Comment:
If this is 0.1s then it might be a little aggressive - if the initial negotiation is a network hiccup then perhaps give it 1s between retries.

Reply:
Good point.  Not sure exactly what is causing the exception (I've yet to see it), but let's give the network a little more time to recover.  I changed it to 1 second.